File: 0070. Climbing Stairs/dp.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n < 0) return 0;
        if(n == 1 || n == 0) return 1;
        
        vector<int> dp(n + 1, 1);
        dp[1] = 1;
        for(int i = 2; i < n + 1; ++i){
            dp[i] = dp[i - 1] + dp[i - 2];
        }

        return dp[n];
    }
};

File: offer/JZ85 连续子数组的最大和(二)/dp.java
import java.util.*;

public class Solution {
    public int[] FindGreatestSumOfSubArray (int[] v) {
        if(v.length == 0)
            return null;
        int len = v.length, ret = v[0], left = 0, right = 0, retLeft = 0, retRight = 0;
        int[] dp = new int[len];
        dp[0] = v[0];
        for(int i = 1; i < len; ++i){
            right++;
            if(dp[i - 1] + v[i] < v[i]){
                left = i;
                dp[i] = v[i];
            }else{
                dp[i] = dp[i - 1] + v[i];
            }
            
            if(dp[i] > ret || (dp[i] == ret && (right - left) > (retRight - retLeft))){
                ret = dp[i];
                retLeft = left;
                retRight = right;
            }
        }
        
        return Arrays.copyOfRange(v, retLeft, retRight + 1);
    }
}